feat(curriculum): add string method workshop (#66936)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-employee-profile-validator/69e6299710319e7cb8ebdc37.md

---
id: 69e6299710319e7cb8ebdc37
title: Step 7
challengeType: 20
dashedName: step-7
---

# --description--

Real-world data is often messy with extra spaces and inconsistent casing.

> Create a variable `messy_name` and assign it `'  sARaH dAVis  '`. Then print it like this:

Reviewer: ```suggestion
Create a variable `messy_name` and assign it the string`  sARaH dAVis  `. Pay attention to the two leading spaces and the two trailing spaces. Then print it like this:
```

```python
print("Original name: '" + messy_name + "'")
```

You will clean this up in the coming steps.

# --hints--

You should create a variable named `messy_name`.

```js
({ test: () => assert(runPython(`_Node(_code).has_variable('messy_name')`)) })
```

You should assign <code>'&nbsp;&nbsp;sARaH dAVis&nbsp;&nbsp;'</code> to `messy_name`.

```js
({ test: () => assert(runPython(`
node = _Node(_code).find_variable('messy_name')
not node.is_empty() and node.is_equivalent("messy_name = '  sARaH dAVis  '")
`)) })
```

> You should print the original name with the prefix `'Original name: '`.

Reviewer: ```suggestion
You should have `print("Original name: '" + messy_name + "'")` in your code.
```

---
id: 69e6299710319e7cb8ebdc37
title: Step 7
challengeType: 20
dashedName: step-7
---

# --description--

Real-world data is often messy with extra spaces and inconsistent casing.

Create a variable `messy_name` and assign it `'  sARaH dAVis  '`. Then print it like this:

> ```python

Reviewer 2: The workshop uses `py` for its fences everywhere else.

````suggestion
```py
````

print("Original name: '" + messy_name + "'")
```

You will clean this up in the coming steps.

# --hints--

You should create a variable named `messy_name`.

```js
({ test: () => assert(runPython(`_Node(_code).has_variable('messy_name')`)) })
```

You should assign <code>'&nbsp;&nbsp;sARaH dAVis&nbsp;&nbsp;'</code> to `messy_name`.

```js
({ test: () => assert(runPython(`
node = _Node(_code).find_variable('messy_name')
not node.is_empty() and node.is_equivalent("messy_name = '  sARaH dAVis  '")
`)) })
```

> You should print the original name with the prefix `'Original name: '`.

Reviewer 2: Stray double blank line between the hints.

```suggestion

You should print the original name with the prefix `'Original name: '`.
```

---
id: 69e6299710319e7cb8ebdc37
title: Step 7
challengeType: 20
dashedName: step-7
---

# --description--

Real-world data is often messy with extra spaces and inconsistent casing.

> Create a variable `messy_name` and assign it the string `  sARaH dAVis  `. Pay attention to the two leading spaces and the two trailing spaces. Then print it so that the output reads:

Reviewer 2: "Create a variable named X" is used in steps 5, 6, 8, 11 and 20, and "Create a variable X" in steps 7, 10, 14, 15, 17 and 18. Every hint in the workshop says "You should create a variable named X", so the descriptions should match.

The same edit applies to step 10 L26, step 14 L18, step 15 L18, step 17 L20 and step 18 L20, each of which has its own comment.

```suggestion
Create a variable named `messy_name` and assign it the string `  sARaH dAVis  `. Pay attention to the two leading spaces and the two trailing spaces. Then print it so that the output reads:
```

```md
Original name: '  sARaH dAVis  '
```

You will clean this up in the coming steps.

# --hints--

You should create a variable named `messy_name`.

```js
({ test: () => assert(runPython(`_Node(_code).has_variable('messy_name')`)) })
```

> You should assign <code>'&nbsp;&nbsp;sARaH dAVis&nbsp;&nbsp;'</code> to `messy_name`.

Reviewer 2: This hint spells the spaces as `&nbsp;` inside a `<code>` tag, but the next hint in the same file writes the same value as `` `Original name: '  sARaH dAVis  '` `` with literal spaces, and so does the fenced block in the description. A markdown code span preserves interior spaces, and this value starts with a quote character so there is no leading-space trimming, which is why the plain form already works two hints down.

Either form is fine as long as the workshop picks one. Step 10 L45 has the same split.

```suggestion
You should assign `'  sARaH dAVis  '` to `messy_name`.
```

---
id: 69e6299710319e7cb8ebdc37
title: Step 6
challengeType: 20
dashedName: step-6
---

# --description--

> Real-world data is often messy with extra spaces and inconsistent casing.

Reviewer: Also here, we should introduce only one new method per step.
